apps/io-lollipop/src/utils/blob.ts

```diff
@@ -0,0 +1,168 @@
+import { BlobServiceClient } from "@azure/storage-blob";
+import { pipe } from "fp-ts/lib/function";
+import * as E from "fp-ts/Either";
+import * as O from "fp-ts/Option";
+import * as TE from "fp-ts/TaskEither";
+import {
+  InternalError,
+  NotFoundError,
+  toInternalError,
+  toNotFoundError
+} from "./errors";
+
+/**
+ * Converts a ReadableStream into a string.
+ *
+ * @param readable - The ReadableStream to convert.
+ * @returns A TaskEither that resolves to the string content of the stream,
+ *          or an InternalError on failure.
+ */
+export const streamToText = (
+  readable: NodeJS.ReadableStream
+): TE.TaskEither<InternalError, string> =>
+  pipe(
+    TE.tryCatch(async () => {
+      readable.setEncoding("utf8");
+
+      // eslint-disable-next-line functional/no-let
+      let result = "";
+      for await (const chunk of readable as AsyncIterable<string | Buffer>) {
+        result += chunk.toString();
+      }
+      return result;
+    }, E.toError),
+    TE.mapLeft(error =>
+      toInternalError(
+        `Unable to download assertion blob: ${error.message}`,
+        `Unable to download assertion blob`
```

Comment:
I suggest to change the name for this error to differentiate from the one in `downloadBlob` function